components/test_raftstore/src/transport_simulate.rs

```diff
@@ -527,6 +527,50 @@ impl Filter for DropSnapshotFilter {
     }
 }
 
+/// Filters all `filter_type` packets until seeing the `flush_type`.
+///
+/// The first filtered message will be flushed too.
+pub struct LeadingFilter {
+    filter_type: MessageType,
+    flush_type: MessageType,
+    first_filtered_msg: Mutex<Option<RaftMessage>>,
+}
+
+impl LeadingFilter {
+    pub fn new(filter_type: MessageType, flush_type: MessageType) -> LeadingFilter {
+        LeadingFilter {
+            filter_type,
+            flush_type,
+            first_filtered_msg: Mutex::default(),
+        }
+    }
+}
+
+impl Filter for LeadingFilter {
+    fn before(&self, msgs: &mut Vec<RaftMessage>) -> Result<()> {
+        let mut filtered_msg = self.first_filtered_msg.lock().unwrap();
+        let mut to_send = vec![];
+        for msg in msgs.drain(..) {
+            if msg.get_message().get_msg_type() == self.filter_type {
+                if filtered_msg.is_none() {
+                    *filtered_msg = Some(msg);
+                }
+            } else if msg.get_message().get_msg_type() == self.flush_type {
```

Comment:
Doesn't not need to `break` here?

Reply:
Following messages may still apply the rules, it may not be appropriate to break as a general filter.